tests/service/ProjectAnalysisTests.fs

```diff
@@ -5462,6 +5462,67 @@ type A(i:int) =
     | Some decl -> failwithf "unexpected declaration %A" decl
     | None -> failwith "declaration list is empty"
 
+[<Test>]
+[<TestCase true>]
+[<TestCase false>]
+let ``TryGetRecentCheckResultsForFile called with snapshot returns cached result after ParseAndCheckFile`` useTransparentCompiler =
```

Comment:
Might be useful to extend the test to include the scenario when some _other_ file is edited but we still want to get cached results for _our file_ even though it's technically invalidated. If that's indeed what should happen. 

(It _might_ be easier to use `ProjectWorkflowBuilder` for it. Or at least the test might be more readable.)

Reply:
Thanks, yeah, will do.